src/Adapter/Assets/AssetUrlGeneratorTrait.php

```diff
@@ -40,8 +40,11 @@ protected function getUriFromPath($fullPath)
 
     protected function getPathFromUri($fullUri)
     {
-        if ('' !== ($trimmedUri = rtrim($this->configuration->get('__PS_BASE_URI__'), '/'))) {
-            return $this->configuration->get('_PS_ROOT_DIR_').preg_replace('/\\'.$trimmedUri.'/', '', $fullUri, 1);
+        $trimmedUri = rtrim($this->configuration->get('__PS_BASE_URI__'), '/');
+
+        if ('' !== $trimmedUri) {
+            $uri = str_replace('/', '\\/', $trimmedUri);
+            return $this->configuration->get('_PS_ROOT_DIR_').preg_replace('/'.$uri.'/', '', $fullUri, 1);
```

Comment:
Thanks for poiting this line. It seems we can go further by replacing preg_replace with a single str_replace, less eager.

```php
        if ('' !== ($trimmedUri = rtrim($this->configuration->get('__PS_BASE_URI__'), '/'))) {
            return $this->configuration->get('_PS_ROOT_DIR_').str_replace($trimmedUri, '', $fullUri);
        }
```
However, I haven't tested on a Windows environment yet.